components/editor/editor.mjs

```diff
@@ -26,10 +27,11 @@ class CMEditor extends HTMLElement {
 
         function update(vu) {
             self.documents[self.currentDocument] = vu.state;
+            window.ide_dispatch_event("save-unsaved-file", {file_name: self.currentDocument, content: vu.state.doc.toString()});
```

Comment:
You can fire the event only if uv.some attribute says the editor content has changed.

Reply:
Thanks! done in https://github.com/FifthTry/ui/pull/89/commits/77eeb06aaec92ce211e0cbed96754fc7b0880ee8